### ISSUE TITLE
Storysource: syntax highlight breaks when typing stories

### ISSUE THREAD

**Issue by petermikitsh** [storybookjs/storybook#12141]
**Describe the bug**

Syntax highlighting in the 'Story' tab breaks when adding a type to a story.

**To Reproduce**

1. Setup a typescript react storybook
2. Add a typescript type to a story

**Expected behavior**

Syntax highlighting works as expected

**Screenshots**

<img width="1076" alt="Screen Shot 2020-08-19 at 9 56 21 AM" src="https://user-images.githubusercontent.com/1571918/90666865-dc2d7500-e202-11ea-9938-8a743b13f662.png">
<img width="1078" alt="Screen Shot 2020-08-19 at 9 56 44 AM" src="https://user-images.githubusercontent.com/1571918/90666874-df286580-e202-11ea-9737-d332247ad3ed.png">

**Code snippets**

The change that causes the bug:

```diff
- export const Default = (args) => <Avatar {...args} />;
+ export const Default: Story<AvatarProps> = (args) => <Avatar {...args} />;
Default.args = {
  label: 'D',
};
```

**System:**

```
  System:
    OS: macOS Mojave 10.14.6
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
  Binaries:
    Node: 12.18.0 - ~/.nvm/versions/node/v12.18.0/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.4 - ~/.nvm/versions/node/v12.18.0/bin/npm
  Browsers:
    Safari: 12.1.2
  npmPackages:
    @storybook/addon-actions: ^6.0.5 => 6.0.13 
    @storybook/addon-essentials: ^6.0.5 => 6.0.13 
    @storybook/addon-links: ^6.0.5 => 6.0.13 
    @storybook/addon-storyshots: ^6.0.12 => 6.0.13 
    @storybook/addon-storyshots-puppeteer: ^6.0.12 => 6.0.13 
    @storybook/addon-storysource: ^6.0.5 => 6.0.13 
    @storybook/addons: ^6.0.5 => 6.0.13 
    @storybook/react: ^6.0.5 => 6.0.13 
```

**Comment by shilman** [comment]
@petermikitsh what's broken about the second screenshot?

**Comment by petermikitsh** [comment]
The keywords `export const` on lines 18 and 24 in the second screenshot are not blue.

**Comment by petermikitsh** [comment]
<img width="607" alt="Screen Shot 2020-09-13 at 5 51 28 PM" src="https://user-images.githubusercontent.com/1571918/93033132-ca64a500-f5e9-11ea-838f-b0779389ee7d.png">

Another example of the syntax highlighting issue: this multi-line comment is partially green and black text.

**Comment by ndelangen** [comment]
This is fixed in 7.0 beta
